src/Build/Definition/ToolsetConfigurationReader.cs

```diff
@@ -90,7 +90,7 @@ protected override IEnumerable<ToolsetPropertyDefinition> ToolsVersions
                             toolset.ElementInformation.LineNumber,
                             0);
 
-                        if (toolset.toolsVersion?.Length == 0)
+                        if (toolset.toolsVersion == null || toolset.toolsVersion.Length == 0)
```

Comment:
This is a behavior change. Before it was "if it is empty but not null"; now null and empty are treated the same. Many of these were that way, and that's why at least some of the tests are failing.